lotus-storage-miner crashed after report a error

when lotus-storage-miner running, occur a error as follows, then crashed.

```2019-12-03T14:00:40.634+0800    WARN    storageminer    storage/fpost_sched.go:125      Aborting Fallback PoSt (EPS: 602)
2019-12-03T14:00:42.043+0800    INFO    storageminer    storage/miner.go:167    Generate candidates took 13.015032ms
2019-12-03T14:00:52.614+0800    INFO    storageminer    storage/fpost_run.go:72 submitting PoSt {"pLen": 192, "elapsed": 17.363529753}
2019-12-03T14:00:52.623+0800    INFO    storageminer    storage/fpost_run.go:137        Submitted fallback post: bafy2bzacebohfywtvhosijlbllr7h47ozhcx43l75c47dr7ofhsgwb75y644y
2019-12-03T14:01:12.026+0800    INFO    storageminer    storage/miner.go:167    Generate candidates took 7.608335ms
2019-12-03T14:01:42.025+0800    INFO    storageminer    storage/miner.go:167    Generate candidates took 7.088825ms
2019-12-03T14:02:12.029+0800    INFO    storageminer    storage/miner.go:167    Generate candidates took 7.027387ms
2019-12-03T14:02:27.059+0800    INFO    storageminer    storage/miner.go:167    Generate candidates took 8.609053ms
^C2019-12-03T14:02:28.922+0800  WARN    main    lotus-storage-miner/run.go:137  Shutting down..
2019-12-03T14:02:42.087+0800    INFO    storageminer    storage/miner.go:167    Generate candidates took 6.944822ms
2019-12-03T14:02:42.089+0800    WARN    deals   deals/provider.go:152   quitting deal provider loop
2019-12-03T14:02:42.089+0800    WARN    storageminer    storage/sealing.go:179  quitting deal provider loop
2019-12-03T14:02:42.090+0800    WARN    events  events/events.go:104    listenHeadChanges quit
2019-12-03T14:02:42.090+0800    WARN    events  events/events.go:107    not restarting listenHeadChanges: context error: context canceled
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xcdaf6f]
goroutine 262 [running]:
github.com/filecoin-project/lotus/chain/types.(*TipSet).Height(...)
        /home/lotus/lotus/chain/types/tipset.go:131
github.com/filecoin-project/lotus/storage.(*fpostScheduler).shouldFallbackPost(0xc0005000f0, 0x1c59a20, 0xc000289b90, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/lotus/lotus/storage/fpost_sched.go:137 +0x11f
github.com/filecoin-project/lotus/storage.(*fpostScheduler).update(0xc0005000f0, 0x1c59a20, 0xc000289b90, 0x0, 0x0, 0x0)
        /home/lotus/lotus/storage/fpost_sched.go:100 +0x4d
github.com/filecoin-project/lotus/storage.(*fpostScheduler).run(0xc0005000f0, 0x1c59960, 0xc0005ce900)
        /home/lotus/lotus/storage/fpost_sched.go:72 +0x299
created by github.com/filecoin-project/lotus/storage.(*Miner).Run
        /home/lotus/lotus/storage/miner.go:111 +0x1db

--- a/storage/fpost_sched.go
+++ b/storage/fpost_sched.go
@@ -49,10 +49,17 @@ func (s *fpostScheduler) run(ctx context.Context) {
 		panic(err)
 	}
 
+	defer s.abortActivePoSt()
+
 	// not fine to panic after this point
 	for {
 		select {
-		case changes := <-notifs:
+		case changes, ok := <-notifs:
+			if !ok {
+				log.Warn("fpostScheduler notifs channel closed")
+				return
+			}
+
 			ctx, span := trace.StartSpan(ctx, "fpostScheduler.headChange")
 
 			var lowest, highest *types.TipSet = s.cur, nil
@@ -74,6 +81,8 @@ func (s *fpostScheduler) run(ctx context.Context) {
 			}
 
 			span.End()
+		case <-ctx.Done():
+			return
 		}
 	}
 }

